AK/Platform.h

```diff
@@ -40,6 +40,21 @@
 #endif
 #define FLATTEN __attribute__((flatten))
 
+#ifdef RETURNS_NONNULL
+#    undef RETURNS_NONNULL
+#endif
+#define RETURNS_NONNULL __attribute__((returns_nonnull))
+
+#ifdef LIKELY
+#    undef LIKELY
+#endif
+#define LIKELY(expr) __builtin_expect(static_cast<bool>(expr), true)
+
+#ifdef UNLIKELY
+#    undef UNLIKELY
+#endif
+#define UNLIKELY(expr) __builtin_expect(static_cast<bool>(expr), false)
```

Comment:
there are `[[likely]]` and `[[unlikely]]` in the spec and hav support from both gcc and clang,
to which are these different than slapping one of them after the conditional?

I guess there can be more fine grained, though there arent many cases where you have mixed likability

Reply:
Unfortunately, they cannot be used with the ternary operator which we use for `VERIFY`.

As Ali suggested, I'll probably just remove these macros altogether. My initial idea was to allow making these macros into no-ops if a compiler does not support `__builtin_expect`, but I guess we won't be compiling with MSVC anytime soon, so there is no need to worry about that.

Reply:
If I am reading it correctly  they actually might,
atleast cppreference.com uses it like that, if I am not mistaken 

https://en.cppreference.com/w/cpp/language/attributes/likely

other wise can you make it an if, or do line numbers change from that?
```cpp
do {
  if (!(expr)) [[unlikely]]
    __assertion_failed(#expr, __FILE__, __LINE__, __PRETTY_FUNCTION__);
} while (0)
```